app/Http/Controllers/TestController.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Validator;
+use App\Test;
+use App\TestQuestion;
+use Illuminate\Http\Request;
+use App\Rules\ValidTestQuestionsScore;
+use App\Http\Resources\Test as TestAndQuestions;
+
+class TestController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $tests = Test::paginate(parent::PER_PAGE);
+
+        return TestAndQuestions::collection($tests);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        Validator::make($request->all(), [
+            'subject_id' => 'required|exists:subjects,id',
+            'group_id' => 'required|exists:groups,id',
+            'start' => 'required|date|after:now',
+            'end' => 'required|date|after:start',
+            'description' => 'string|max:65000|nullable',
+            'title' => 'string|max:255',
+            'questions' => [
+                'required', 'array', new ValidTestQuestionsScore
+            ],
+            'questions.*.score' => 'required|numeric',
+            'questions.*.question_id' => 'required|exists:questions,id',
+        ])->validate();
+      
+        $test = Test::create([
+            'subject_id' => $request->subject_id,
+            'group_id' => $request->group_id,
+            'start' => $request->start,
+            'end' => $request->end,
+            'title' => $request->title,
+            'description' => $request->description,
+        ]);
+
+        foreach ($request->questions as $question) {
+            $questions[] = TestQuestion::make($question);
+        }
+
+        $test->questions()->saveMany($questions);
```

Comment:
We have a belongsToMany relation, which records should be attached with the **attach()** method.
Read about relations in the laravel documentation.
https://laravel.com/docs/7.x/eloquent-relationships